test/application.test.js

```diff
@@ -87,3 +87,35 @@ test('Register express application that uses Router', t => {
     })
   })
 })
+
+test('Should allow removal of x-powered-by header', t => {
```

Comment:
Can you split this into two tests, one for the disable and one for the new decorator?

Reply:
should I split into 2 PRs? One fixing the broken removal of the powered by header, and one adding the decorator?

Reply:
Whatever you prefer, I'm good either way.

Reply:
pushed just the fix for removal of the header now. PR with decorator incoming